### PR TITLE
feat(ui): popover

### PR DESCRIPTION
**Pull request by neSpecc**
Popover component added.

1. Popover — the empty wrapper that can contain any component
2. It can be shown near a particular target element
3. Can be aligned vertically and horizontally 
4. Can be stretched to full width of a target element
5. Hides on clicks outside.

image

### CODE REVIEW

File: codex-ui/src/vue/components/popover/Popover.types.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+export interface PopoverShowParams {
+  /**
+   * Element to move popover to
+   */
+  targetEl: HTMLElement;
+
+  /**
+   * Popover content: component and props
+   */
+  with: PopoverContent;
```
Comment (e11sy) [inline codex-ui/src/vue/components/popover/Popover.types.ts:31]: maybe just content?

File: codex-ui/src/vue/components/popover/usePopover.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    isOpen.value = false;
+  }
+
+  /**
+   * Hides and resets the popover
+   */
+  function hide(): void {
+    resetPopover();
+  }
+
```
Comment (e11sy) [inline codex-ui/src/vue/components/popover/usePopover.ts:154]: seems redundant

File: src/App.vue [hunk 3]
```diff
@@ -1,12 +1,13 @@
 <template>
   <Header />
   <router-view />
+  <Popover />
```
Comment (e11sy) [inline src/App.vue:4]: does it used?

File: codex-ui/src/vue/components/popover/Popover.vue [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+import { usePopover } from './usePopover';
+import { onClickOutside } from '@vueuse/core';
+
+const popoverEl = ref<HTMLDivElement | null>(null);
```
Comment (elizachi) [inline codex-ui/src/vue/components/popover/Popover.vue:20]: Add comment please

File: codex-ui/dev/Playground.vue [hunk 5]
```diff
@@ -29,14 +29,16 @@
         <router-view />
       </div>
     </div>
+    <Popover />
```
[the remark this answers is not in the document]
Comment (neSpecc) [reply codex-ui/dev/Playground.vue:32]: It is required to add Popover component to the base app vue. We have the single popover for the app and it is managed by the `usePopover` composable

File: codex-ui/src/vue/components/popover/usePopover.ts (hunk 2 above)
Comment (neSpecc) [reply codex-ui/src/vue/components/popover/usePopover.ts:154]: it is used to hide popover, for clicking outside of it for example. Or manually

File: codex-ui/src/vue/components/popover/Popover.types.ts (hunk 1 above)
Comment (neSpecc) [reply codex-ui/src/vue/components/popover/Popover.types.ts:31]: its too generic

**Review by e11sy — APPROVED** [review]

**Review by elizachi — APPROVED** [review]

**Pull request merged**